subprojects/packagefiles/libarchive/meson.build

```diff
@@ -107,6 +107,12 @@ if cc.has_header_symbol('errno.h', 'EILSEQ')
   cdata.set('HAVE_EILSEQ', 1)
 endif
 
+foreach w : ['_mkgmtime', 'ctime_s', 'gmtime_s', 'localtime_s']
+  if cc.has_header_symbol('time.h', w)
+    cdata.set('HAVE_@0@'.format(w.to_upper()), 1)
+  endif
+endforeach
+
```

Comment:
Maybe move the check for `gmtime_r` here too?

Reply:
No need. I moved the code upstream to try the _s versions first and then the _r. _r is implemented in terms of _s as a macro IIRC. _r on POSIX systems is available as a function.